website/project/views/comment.py

```diff
@@ -1,21 +1,54 @@
 # -*- coding: utf-8 -*-
 import pytz
+from datetime import datetime
 from flask import request
 from modularodm import Q
+from modularodm.exceptions import NoResultsFound
 
 from framework.auth.decorators import must_be_logged_in
+from framework.tasks import app
 
+from website.files.models import FileNode, TrashedFileNode
 from website.notifications.constants import PROVIDERS
 from website.notifications.emails import notify
 from website.models import Comment
-from website.project.decorators import must_be_contributor_or_public
-from website.project.signals import comment_added, file_moved_node
-from datetime import datetime
+from website.project.decorators import must_be_contributor_or_public, must_have_permission
+from website.project.model import Node
+from website.project.signals import comment_added
+from website import settings
+
+
+@app.task
```

Comment:
Even if it does work, you aren't calling the function asynchronously (e.g. with `my_task.delay`), so the `@app.task` doesn't actually do anything.

To fix this, I suggest the approach in https://github.com/CenterForOpenScience/osf.io/pull/4901#issuecomment-172059266